source/frontend/src/features/notes/models.ts

```diff
@@ -12,31 +13,32 @@ export class NoteForm {
   appLastUpdateUserGuid?: string;
   appCreateUserGuid?: string;
 
-  static fromApi(base: Api_Note): NoteForm {
+  static fromApi(base: ApiGen_Concepts_Note): NoteForm {
     var model = new NoteForm();
     model.id = base.id;
-    model.note = base.note;
-    model.rowVersion = base.rowVersion;
+    model.note = base.note ?? undefined;
+    model.rowVersion = base.rowVersion ?? undefined;
     model.appCreateTimestamp = base.appCreateTimestamp;
-    model.appCreateUserGuid = base.appCreateUserGuid;
-    model.appCreateUserid = base.appCreateUserid;
+    model.appCreateUserGuid = base.appCreateUserGuid ?? undefined;
+    model.appCreateUserid = base.appCreateUserid ?? undefined;
     model.appLastUpdateTimestamp = base.appLastUpdateTimestamp;
-    model.appLastUpdateUserGuid = base.appLastUpdateUserGuid;
-    model.appLastUpdateUserid = base.appLastUpdateUserid;
+    model.appLastUpdateUserGuid = base.appLastUpdateUserGuid ?? undefined;
+    model.appLastUpdateUserid = base.appLastUpdateUserid ?? undefined;
     return model;
   }
 
-  toApi(): Api_Note {
+  toApi(): ApiGen_Concepts_Note {
     return {
-      id: this.id,
-      note: this.note,
-      rowVersion: this.rowVersion,
-      appCreateTimestamp: this.appCreateTimestamp,
-      appCreateUserGuid: this.appCreateUserGuid,
-      appCreateUserid: this.appCreateUserid,
-      appLastUpdateTimestamp: this.appLastUpdateTimestamp,
-      appLastUpdateUserGuid: this.appLastUpdateUserGuid,
-      appLastUpdateUserid: this.appLastUpdateUserid,
+      id: this.id ?? 0,
+      note: this.note ?? null,
+      isSystemGenerated: false,
```

Comment:
should we not pass the value that we have?

Reply:
it should. but at this point, the form does not contain that information.